contracts/timelock/timelock.go

```diff
@@ -0,0 +1,138 @@
+/*
+Copyright 2023 The Bestchains Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package timelock
+
+import (
+	"encoding/json"
+	"fmt"
+	"hash/fnv"
+	"strconv"
+	"time"
+
+	"github.com/bestchains/bestchains-contracts/library/context"
+	"github.com/bestchains/bestchains-contracts/library/timer"
+	"github.com/hyperledger/fabric-contract-api-go/contractapi"
+)
+
+// Event is an operation:
+// whose detail is {Payload},
+// fnv-1 hashed as {Hash},
+// set to be available after {TimeStamp}
+type Event struct {
+	Timestamp timer.TimeStamp
+	Payload   []byte
+	OpType    string
+}
+
+var _ ITimeLock = new(TimeLock)
+
+// TimeLock provides simple key-value Get/Put with time lock function as a usage example
+type TimeLock struct {
+	contractapi.Contract
+}
+
+// Schedule sets a time lock for a {key}-{value} pair, which will be released for any get/set/del after {duration}.
+func (tlc *TimeLock) Schedule(ctx context.ContextInterface, key string, value string, duration int64) (string, error) {
+
+	// Input check
+	if key == "" || value == "" {
+		return "", fmt.Errorf("input error")
+	}
+
+	if duration < 0 {
+		return "", fmt.Errorf("duration cannot be negative")
+	}
+
+	// Form a dictionary entry
+	payload, err := json.Marshal(Entry{
+		Key:   key,
+		Value: value,
+	})
+	if err != nil {
+		return "", err
+	}
+
+	// Emit an event to be saved
+	newEvent := Event{
+		OpType:    "put",
+		Timestamp: timer.TimeStamp{Deadline: time.Now().Unix() + duration},
```

Comment:
it's better not export `Deadline` operation. should use your pre-defined function `SetDeadline`